pkg/kubectl/cmd/apply.go

```diff
@@ -343,18 +343,18 @@ func (p *pruner) prune(namespace string, mapping *meta.RESTMapping, shortOutput
 
 func (p *pruner) delete(namespace, name string, mapping *meta.RESTMapping, c resource.RESTClient) error {
 	if !p.cascade {
-		if err := resource.NewHelper(c, mapping).Delete(namespace, name); err != nil {
-			return err
-		}
-		return nil
+		return resource.NewHelper(c, mapping).Delete(namespace, name)
 	}
 	cs, err := p.clientsetFunc()
 	if err != nil {
 		return err
 	}
 	r, err := kubectl.ReaperFor(mapping.GroupVersionKind.GroupKind(), cs)
 	if err != nil {
-		return err
+		if _, ok := err.(*kubectl.NoSuchReaperError); !ok {
+			return err
+		}
+		return resource.NewHelper(c, mapping).Delete(namespace, name)
```

Comment:
Optional but maybe add a warning here for falling back to non cascading deletion?